static/js/publisher/release/components/defaultTrackModifier.js

```diff
@@ -0,0 +1,196 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import { getTracks, getTrackRevisions } from "../selectors";
+import { CLOSE_MODAL, openModal, closeModal } from "../actions/modal";
+import { showNotification, hideNotification } from "../actions/notification";
+
+class DefaultTrackModifier extends Component {
+  constructor(props) {
+    super(props);
+
+    this.setDefaultTrackHandler = this.setDefaultTrackHandler.bind(this);
+    this.clearDefaultTrackHandler = this.clearDefaultTrackHandler.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      defaultTrack,
+      currentTrack,
+      closeModal,
+      showNotification,
+      snapName
+    } = this.props;
+    if (defaultTrack.track !== prevProps.defaultTrack.track) {
+      closeModal();
```

Comment:
Why does it happen here?

We have `setDefaultTrack` action. It could dispatch any actions on success - including closing modal and even more accurately success messages (that should be direct results of fetch, not this component).